evalml/automl/automl_search.py

```diff
@@ -425,6 +427,7 @@ def search(self, X, y, data_checks="auto", show_iteration_plot=True):
         if self.allowed_pipelines == []:
             raise ValueError("No allowed pipelines to search")
         if self.max_batches and self.max_iterations is None:
+            self.show_batch_output = True
```

Comment:
@christopherbunn what's this? Why do we need it?

Reply:
Currently, we show the batch number only if the user specifies a number of `max_batches`. Since we use batching internally even if only `max_iterations` is specified, there isn't really a clean way to infer whether or not we want to show the number of batches other than setting a variable at the beginning.